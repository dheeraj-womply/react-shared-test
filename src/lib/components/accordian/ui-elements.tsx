import styled from "styled-components";
import { AccordionSummary, AccordionDetails } from '@material-ui/core';
import { DEFAULT_FONT_FAMILY_BOLD } from '../../themes/default-styles';

export const AccordianTitle = styled.div`
    font-size: 20px;
    font-weight: 500;
    letter-spacing: .005em;
    font-family: ${DEFAULT_FONT_FAMILY_BOLD};
`;

export const StyledAccordionSummary = styled(AccordionSummary)`
    padding: 0;
`;

export const StyledAccordionDetails = styled(AccordionDetails)`
    padding: 0;
`;