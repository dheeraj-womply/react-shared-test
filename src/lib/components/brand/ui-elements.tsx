import React from 'react'
import styled from 'styled-components'
import { WOMPLY_LOGO, WOMPLY_LOGO_RED } from '../../resources/images'

export const LogoWrapper = styled.div`
    padding: 20px;
    img {
        width: 137px;
        height: 27px;
    }
`
interface LogoProps {
    alt?: string
    variant?: 'default' | 'red'
}

export const Logo: React.FC<LogoProps> = ({ alt, variant = 'default' }) => (
    <LogoWrapper>
        <img src={variant === 'red' ? WOMPLY_LOGO_RED : WOMPLY_LOGO} alt={alt}/>
    </LogoWrapper>
)
