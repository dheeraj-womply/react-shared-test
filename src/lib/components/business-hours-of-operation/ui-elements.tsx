import styled from 'styled-components';
import { DEFAULT_FONT_FAMILY, MEDIUM_FONT_SIZE } from '../../themes/default-styles';
import { Grid, Button } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';

export const BusinessHoursWrapper = styled.div`
    font-family: ${DEFAULT_FONT_FAMILY};
    font-size: ${MEDIUM_FONT_SIZE};
`;

export const DailyHoursWrapper = styled(Grid)`
    padding: 8px 0;
    font-family: ${DEFAULT_FONT_FAMILY};
`;

export const StyledEditIcon = styled(EditIcon)`
    width: 18px;
    vertical-align: bottom;
    color: ${(props: {customcolor: string, float: string}) => props.customcolor};
    float: ${(props: {float: string}) => props.float};
`;

export const TextGrid = styled(Grid)`
    text-align: ${(props: {textalign: string, color: string}) => props.textalign};
    color: ${(props: {color: string}) => props.color};
`;

export const TimepickerFieldLabel = styled.p`
    margin-bottom: 0;
    font-size: ${MEDIUM_FONT_SIZE};
`;

export const TimePickerIconGrid = styled(Grid)`
    margin: auto !important;
    cursor: ${(props: {cursor: string}) => props.cursor};
`;

export const TimePickerIcon = styled.img`
    width: 20px; 
    opacity: 0.6;
`;

export const PopoverButtonGrid = styled(Grid)`
    margin: 15px auto !important;
`;

export const PopoverButton = styled(Button)`
    min-width: 100px;
    font-family: ${DEFAULT_FONT_FAMILY};
`;