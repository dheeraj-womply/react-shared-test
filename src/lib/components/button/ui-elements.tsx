import React from "react";
import {
    Button as MuiButton, ButtonProps,
    makeStyles,
} from '@material-ui/core'

const useStyles = makeStyles({
    root: {
        fontSize: 14,
        fontWeight: 700,
        textTransform: 'uppercase',
    },
})

export const Button: React.FC<ButtonProps> = (props) => {
    const {
        color = 'primary',
        size = 'large',
        variant = 'contained',
        startIcon,
        ...rest
    } = props

    // eslint-disable-next-line react-hooks/rules-of-hooks
    const classes = props.classes ? { ...useStyles(), ...props.classes } : useStyles()

    return (
        <MuiButton
            {...rest}
            color={color}
            size={size}
            startIcon={startIcon}
            variant={variant}
            classes={classes}
        />
    )
}



