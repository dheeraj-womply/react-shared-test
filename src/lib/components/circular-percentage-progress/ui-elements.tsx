import React from 'react'
import { Box, CircularProgress, CircularProgressProps } from '@material-ui/core'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import styled from 'styled-components'

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        bottom: {
            color: theme.palette.grey[theme.palette.type === 'light' ? 200 : 700]
        },
        top: {
            animationDuration: '550ms',
            position: 'absolute',
            left: 0
        },
        circle: {
            strokeLinecap: 'round'
        }
    })
)

const ProgressText = styled.span`
  font-size: 12px;
  font-weight: bold;
`

const ableToBeNormalized = (value: any): boolean => {
    if (['number', 'string'].includes(typeof value)) {
        const val = +value
        return val === 0 || (Math.round(val) > 0 && Math.round(val) <= 100)
    }
    return true
}

const normalize = (value: any): number => {
    if (['number', 'string'].includes(typeof value)) {
        return Math.round(+value)
    }
    return 0
}

/**
 * The 'value' prop is expected to be an integer between 0 and 100, where the type is 'number'.
 * If a string is provided and it can be converted to a number between 0 and 100, that is also supported.
 * Any non-integer will be rounded to the nearest integer.
*/
export const CircularPercentageProgress: React.FC<CircularProgressProps> = (props) => {
    const classes = useStyles()

    return (
        <>
        {ableToBeNormalized(props.value) &&
            <Box position="relative" display="inline-flex">
                <Box position="relative" display="flex">
                    <CircularProgress
                        variant="static"
                        size={50}
                        thickness={5}
                        value={100}
                        className={classes.bottom}
                    />
                    <CircularProgress
                        variant="static"
                        size={50}
                        thickness={5}
                        {...props}
                        value={normalize(props.value)}
                        className={classes.top}
                        classes={{
                            circle: classes.circle
                        }}
                    />
                </Box>
                <Box
                    top={0}
                    left={0}
                    bottom={0}
                    right={0}
                    position="absolute"
                    display="flex"
                    alignItems="center"
                    justifyContent="center"
                >
                    <ProgressText>{`${normalize(props.value)}%`}</ProgressText>
                </Box>
            </Box>
        }
        </>
    )
}
