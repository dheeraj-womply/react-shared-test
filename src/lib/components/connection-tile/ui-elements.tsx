import React from 'react'
import Tooltip from '@material-ui/core/Tooltip'
import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight'
import PriorityHighIcon from '@material-ui/icons/PriorityHigh'
import CheckIcon from '@material-ui/icons/Check'
import { IConnectionTile, TileStatus } from './interfaces'
import styled from 'styled-components'
import { PRIMARY_MAIN_COLOR, SECONDARY_MAIN_COLOR, ERROR_MAIN_COLOR, INFO_MAIN_COLOR, DEFAULT_FONT_FAMILY_BOLD } from '../../themes/default-styles'

export const Tile = styled.div`
    border-radius: 3px;
    position: relative;
    border: 1px solid #DDE5E8;
    min-width: 150px;
    &:hover {
        cursor: pointer;
    }
    &:hover svg.arrow {
        fill: ${PRIMARY_MAIN_COLOR};
    }
    &:hover div:nth-of-type(2) span:first-of-type {
        color: ${PRIMARY_MAIN_COLOR};
    }
    &:hover {
        box-shadow: 5px 10px 20px 0 rgba(29, 48, 70, 0.15);
    }
    &.CONNECTION_PENDING {
        border: 2px solid ${INFO_MAIN_COLOR};
    }
    &.CONNECTED_INVALID {
        border: 2px solid ${ERROR_MAIN_COLOR};
    }
    &.CONNECTED_VALID {
        border: 2px solid ${SECONDARY_MAIN_COLOR};
    }
`

export const TileIconContainer = styled.div`
    padding: 15px 26px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    height: 70px;
    border-bottom: 1px solid #DDE5E8;
`

export const TileIcon = styled.img`
    max-width: 100%;
    max-height: 69px;
`

export const TileTextContainer = styled.div`
    height: 30px;
    display: flex;
    padding: 0 10px;
    justify-content: space-between;
    align-items: center;
`

export const TileText = styled.span`
    font-family: ${DEFAULT_FONT_FAMILY_BOLD};
    font-size: 12px;
    font-weight: bold;
    letter-spacing: 0;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
`

export const TileImageText = styled.span`
    font-size: 12px;
    font-family: ${DEFAULT_FONT_FAMILY_BOLD};
    text-align: center;
    max-height: 39px;
    line-height: 19.5px;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
`

export const TileTextArrow = styled.span`
    display: flex;
    margin-right: -8px;
`

export const TileStatusContainer = styled.div`
    position: absolute;
    right: -1px;
    top: -1px;
    display: flex;
    padding: 3.5px;
    border-bottom-left-radius: 3px;
    color: #fff;
    font-size: 0.8rem;
    &.CONNECTION_PENDING {
        background: ${INFO_MAIN_COLOR};
    }
    &.CONNECTED_INVALID {
        background: ${ERROR_MAIN_COLOR};
    }
    &.CONNECTED_VALID {
        background: ${SECONDARY_MAIN_COLOR};
    }
`

export const ConnectionTile: React.FC<IConnectionTile> = ({ tile, onClick }) => {
    const getIconBasedOnStatus = (status: TileStatus) => {
        switch (status) {
            case TileStatus.CONNECTED_INVALID:
                return <PriorityHighIcon fontSize="inherit" />
            case TileStatus.CONNECTED_VALID:
                return <CheckIcon fontSize="inherit" />
            case TileStatus.CONNECTION_PENDING:
                return <PriorityHighIcon fontSize="inherit" />
            default:
                return <></>
        }
    }

    return (
        <Tooltip title={tile.label}>
            <Tile className={tile.status} onClick={(event) => onClick && onClick(event)}>
                <TileIconContainer>
                    {
                        !tile.icon &&
                        <TileImageText>{tile.label}</TileImageText>
                    }
                    {
                        tile.icon &&
                        <TileIcon
                            src={tile.icon}
                            alt={tile.label}
                        />
                    }
                </TileIconContainer>
                <TileTextContainer>
                    <TileText>{tile.label}</TileText>
                    <TileTextArrow><KeyboardArrowRightIcon className={'arrow'}/></TileTextArrow>
                </TileTextContainer>
                <TileStatusContainer className={tile.status}>
                    {getIconBasedOnStatus(tile.status)}
                </TileStatusContainer>
            </Tile>
        </Tooltip>
    )
}