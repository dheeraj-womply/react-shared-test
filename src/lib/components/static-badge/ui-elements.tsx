import React from "react";
import {Badge, BadgeProps, Box} from "@material-ui/core";
import {makeStyles} from "@material-ui/core/styles";

type StaticBadgeProps = Pick<BadgeProps, Exclude<keyof BadgeProps, 'color'>> & { color?: string; bgcolor?: string; }

const useStaticBadgeStyles = makeStyles({
    root: (props: StaticBadgeProps) => ({
        color: 'inherit',
        '&& .MuiBadge-badge': {
            color: `${!props.color || props.color === 'default' ? '#99a5aa' : ''}`,
            backgroundColor: `${
                props.bgcolor ||
                (!props.color || props.color === 'default'
                    ? '#f5f6f5'
                    : ['primary', 'secondary', 'error'].includes(props.color) ? '' : '#fff')
            }`,
            position: 'static',
            transform: 'none',
            height: 24,
            minWidth: 24,
            borderRadius: 12
        }
    })
})

export const StaticBadge:React.FC<StaticBadgeProps> = (props) => {
    const classes = useStaticBadgeStyles(props)
    const color = (
        props.color && ['default', 'primary', 'secondary', 'error'].includes(props.color) ?
            props.color as 'default' | 'primary' | 'secondary' | 'error' :
            undefined
    )
    const badgeProps = { ...props, color}

    return (
        <Box color={props.color}>
            <Badge {...badgeProps} className={classes.root} />
        </Box>
    )
}
