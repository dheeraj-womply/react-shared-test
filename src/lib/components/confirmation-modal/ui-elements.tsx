import React from 'react';
import { ConfirmationModalComponent } from './interfaces';
import { Dialog, DialogActions, DialogContent, DialogTitle } from '@material-ui/core';
import {Button} from "../button";


export const ConfirmationModal: React.FC<ConfirmationModalComponent> = (props) => {
    const {
        isModalOpen = false,
        isMobile = false,
        title,
        content,
        primaryButtonText,
        onPrimaryButtonClick,
        secondaryButtonText,
        onSecondaryButtonClick
    } = props

    return (
        <Dialog
            open={isModalOpen}
            maxWidth="xs"
            fullWidth={true}
        >
            <DialogTitle>
                {title}
            </DialogTitle>
            <DialogContent>
                {content}
            </DialogContent>
            <DialogActions>
                <Button
                    data-testid="secondary-btn"
                    color="primary"
                    variant="text"
                    fullWidth={isMobile}
                    onClick={onSecondaryButtonClick}
                    disableRipple
                >
                    {secondaryButtonText}
                </Button>
                <Button
                    data-testid="primary-btn"
                    fullWidth={isMobile}
                    disableRipple
                    variant="contained"
                    color="primary"
                    onClick={onPrimaryButtonClick}
                >
                    {primaryButtonText}
                </Button>
            </DialogActions>
        </Dialog>
    )
}
