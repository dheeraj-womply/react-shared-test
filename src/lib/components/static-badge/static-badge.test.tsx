import React from 'react'
import renderer from 'react-test-renderer'
import {StaticBadge} from "./ui-elements";
import {number} from "@storybook/addon-knobs";

test('Static Badge DOM elements are OK', () => {
    const component = renderer.create(<StaticBadge color="default" badgeContent={100} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('Static Badge with bg color DOM elements are OK', () => {
    const component = renderer.create(<StaticBadge
        badgeContent={number('content-5', 100)}
        max={100}
        bgcolor="purple"
        color="#fff"
    />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

