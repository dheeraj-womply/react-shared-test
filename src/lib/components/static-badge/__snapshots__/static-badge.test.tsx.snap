// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Badge DOM elements are OK 1`] = `
<div
  className="MuiBox-root MuiBox-root-3"
>
  <span
    className="MuiBadge-root makeStyles-root-1 makeStyles-root-2"
  >
    <span
      className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle"
    >
      99+
    </span>
  </span>
</div>
`;

exports[`Static Badge with bg color DOM elements are OK 1`] = `
<div
  className="MuiBox-root MuiBox-root-5"
>
  <span
    bgcolor="purple"
    className="MuiBadge-root makeStyles-root-1 makeStyles-root-4"
  >
    <span
      className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle"
    >
      100
    </span>
  </span>
</div>
`;
